specs/commandline/blob.md

```diff
@@ -0,0 +1,504 @@
+
+User can refer to the below trust policy configuration sample `trustpolicy.blob.json` that is applicable for verifying signed arbitrary blobs using `notation blob verify` command. In this sample, there are three policies configured for different requirements:
+
+- The Policy named "wabbit-networks-policy" is for verifying blob artifacts signed by Wabbit Networks.
+- Policy named "skip-verification-policy" is for skipping verification on blob artifacts.
+- Policy "wildcard-verification-policy" is for auditing verification results when user wants to apply a wildcard policy by not providing `--policy-name` argument in `notation blob verify` command.
+
+```jsonc
+{
+    "version": "1.0",
+    "policyType": "blob",
+    "trustPolicies": [
+        {
+            "name": "wabbit-networks-policy",
+            "signatureVerification": {
+                "level": "strict"
+            },
+            "trustStores": [ 
+              "ca:wabbit-networks",
+            ],
+            "trustedIdentities": [
+                "x509.subject: C=US, ST=WA, L=Seattle, O=wabbit-networks.io, OU=Security Tools"
+            ]
+        },
+        {
+            "name": "skip-verification-policy",
+            "signatureVerification": {
+              "level" : "skip" 
+            }
+        },
+        {
+            "name": "wildcard-verification-policy",
+            "wildcardPolicy": true,
+            "signatureVerification": {
+              "level" : "audit"
+            },
+            "trustStores": ["ca:acme-rockets"],
+            "trustedIdentities": ["*"]
+        }
+    ]
+}
+```
+
+## Outline
+
+### notation blob command
+
+```text
+Sign, inspect, and verify signatures associated with arbitrary blobs.
+
+Usage:
+  notation blob [command]
+
+Available Commands:
+  inspect   inspect a signature associated with a blob
+  policy    manage trust policy configuration file for signed blobs
+  sign      produce a detached signature for a given blob
+  verify    verify a signature associated with a blob
+
+Flags:
+  -h, --help   help for blob
+```
+
+### notation blob sign
+
+```text
+Produce a signature for a given blob. A detached signature file will be written to the currently working directory with blob file name + ".sig" + signature format as the file extension. For example, signature file name for "myBlob.bin" will be "myBlob.bin.sig.jws" for JWS signature format or "myBlob.bin.sig.cose" for COSE signature format.
+
+Usage:
+  notation blob sign [flags] <blob_path>
+
+Flags:
+  --signature-directory string optional path where the blob signature needs to be placed (default: currently working directory) 
```

Comment:
Could we simplify the flag name since it may be used very frequently? For example, `-D`.

Reply:
`-d` conflicts with debug flag. We decided in one of the earlier threads to drop the short-hand flag.